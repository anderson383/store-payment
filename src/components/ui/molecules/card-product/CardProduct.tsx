import { Link } from 'react-router-dom'
import styles from './CardProduct.module.scss'
import { ProductType } from 'types/inventary';
import { formatPrice } from '../../../../helper/formatPrice';

interface CardProductProps {
  to: string;
  product: ProductType
}
export const CardProduct:React.FC<CardProductProps> = ({to, product}) => {

  return  (
    <Link to={to} className={styles.card}>
      <img src={product.images.length ? product.images[0] : ''} alt={product.name} />

      <div className={styles.content}>
        <caption>Tennis</caption>
        <p>{product.name}</p>
      </div>
      <div className={styles.info}>
        <caption>Stock {product.stock}</caption>
        <p>{formatPrice(product.price)}</p>
      </div>
    </Link >
  )
}