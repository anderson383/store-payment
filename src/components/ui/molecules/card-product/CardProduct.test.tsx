
import {
  render,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react';
import { CardProduct } from './CardProduct';

describe('Tests in CardProduct component', () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <CardProduct
          to={'/test'}
          product={{
            description: 'test',
            id: 'test',
            images: [],
            name: 'test',
            price: 0,
            stock: 0
          }}
        />
      </BrowserRouter>
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });
});
