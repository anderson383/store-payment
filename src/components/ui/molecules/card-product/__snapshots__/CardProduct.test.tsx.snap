// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in CardProduct component Should render the component 1`] = `
<DocumentFragment>
  <a
    class="card"
    href="/test"
  >
    <img
      alt="test"
      src=""
    />
    <div
      class="content"
    >
      Tennis
      <p>
        test
      </p>
    </div>
    <div
      class="info"
    >
      Stock 0
      <p>
        $Â 0
      </p>
    </div>
  </a>
</DocumentFragment>
`;
